test/hotel-test.js

```diff
@@ -0,0 +1,186 @@
+
+    expect(hotel.allBookings.length).to.equal(5)
+  });
+
+  it('should have a method to add customers to array', () => {
+    hotel.addCustomers()
+
+    expect(hotel.allCustomers.length).to.equal(2)
+  });
+
+  it('should have a method to filter bookings by customer ID', () => {
+    hotel.addCustomers();
+    hotel.addBookings();
+    
+    expect(hotel.filterBookingsByCustomerID(customer2)).to.be.deep.equal([
+      {
+        id: '5fwrgu4i7k55hl6ty',
+        userID: 50,
+        date: '2022/01/10',
+        roomNumber: 25
+      },
+      {
+        id: '5fwrgu4i7k55hl6xu',
+        userID: 50,
+        date: '2022/02/25',
+        roomNumber: 19
+      },
+      {
+        id: '5fwrgu4i7k55hl71d',
+        userID: 50,
+        date: '2022/02/03',
+        roomNumber: 9
+      },
+      {
+        id: '5fwrgu4i7k55hl77h',
+        userID: 50,
+        date: '2022/01/21',
+        roomNumber: 13
+      }
+    ])
+  });
+  
+  it('should have a method to sum user\'s bookings', () => {
+    hotel.addCustomers();
+    hotel.addBookings();
+    hotel.addRooms();
+
+    expect(hotel.sumUserBookings(customer2)).to.equal(798.59)
+  });
+
+  it('should have a method to filter by date', () => {
+    hotel.addRooms();
+    hotel.addBookings();
+
+    expect(hotel.filterRoomsByDate('2022-02-25')).to.deep.equal([
+      {
+        roomNumber: 1,
+        roomType: 'residential suite',
+        bidet: true,
+        bedSize: 'queen',
+        numBeds: 1,
+        cost: 358.4
+      },
+      {
+        roomNumber: 2,
+        roomType: 'suite',
+        bidet: false,
+        bedSize: 'full',
+        numBeds: 2,
+        cost: 477.38
+      },
+      {
+        roomNumber: 3,
+        roomType: 'single room',
+        bidet: false,
+        bedSize: 'king',
+        numBeds: 1,
+        cost: 491.14
+      },
+      {
+        roomNumber: 11,
+        roomType: 'single room',
+        bidet: true,
+        bedSize: 'twin',
+        numBeds: 2,
+        cost: 207.24
+      },
+      {
+        roomNumber: 13,
+        roomType: 'single room',
+        bidet: false,
+        bedSize: 'queen',
+        numBeds: 2,
+        cost: 423.92
+      }
+    ])
+  });
+
+  it('should have a method to filter rooms available by type', () => {
+    hotel.addRooms();
+    hotel.addBookings();
+
+    expect(hotel.filterRoomsByType('single room', '2022-02-25')).to.deep.equal([
+      {
+        roomNumber: 3,
+        roomType: 'single room',
+        bidet: false,
+        bedSize: 'king',
+        numBeds: 1,
+        cost: 491.14
+      },
+      {
+        roomNumber: 11,
+        roomType: 'single room',
+        bidet: true,
+        bedSize: 'twin',
+        numBeds: 2,
+        cost: 207.24
+      },
+      {
+        roomNumber: 13,
+        roomType: 'single room',
+        bidet: false,
+        bedSize: 'queen',
+        numBeds: 2,
+        cost: 423.92
+      }
+    ])
+  });
+
+  it('should tell you when nothing is available', () => {
```

Comment:
LOVE THIS! I have my 'sorry we don't have any' message in two different functions. This is cleaner. Awesome!